engine/table/src/main/java/io/deephaven/engine/table/impl/RowKeysColumnSource.java

```diff
@@ -0,0 +1,65 @@
+package io.deephaven.engine.table.impl;
+
+import io.deephaven.chunk.WritableChunk;
+import io.deephaven.chunk.WritableLongChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.rowset.TrackingRowSet;
+import org.apache.commons.lang3.mutable.MutableLong;
+import org.jetbrains.annotations.NotNull;
+
+public class RowKeysColumnSource extends AbstractColumnSource<Long> implements MutableColumnSourceGetDefaults.ForLong {
+
+    private final TrackingRowSet rowSet;
+
+    public RowKeysColumnSource(final TrackingRowSet rowSet) {
+        super(Long.class);
+        this.rowSet = rowSet;
+    }
+
+    @Override
+    public long getLong(long rowKey) {
+        return rowSet.get(rowKey);
+    }
+
+    @Override
+    public long getPrevLong(long rowKey) {
+        return rowSet.getPrev(rowKey);
+    }
+
+    @Override
+    public void fillChunk(@NotNull FillContext context, @NotNull WritableChunk<? super Values> destination,
+            @NotNull RowSequence rowSequence) {
+        final WritableLongChunk<? super Values> longChunk = destination.asWritableLongChunk();
+        longChunk.setSize(0);
+
+        final MutableLong position = new MutableLong();
+        final RowSequence.Iterator rsit = rowSet.getRowSequenceIterator();
+        rowSequence.forAllRowKeys(rowKey -> {
+            long numToAdvance = rowKey - position.longValue();
+            if (numToAdvance > 0) {
+                rsit.getNextRowSequenceWithLength(numToAdvance);
+                position.add(numToAdvance);
+            }
+            longChunk.add(rsit.peekNextKey());
+        });
```

Comment:
We can replace this with `rowSet.invert(rowSequence.asRowSet())` and fill the destination chunk from the result, with appropriate closing. (I don't remember if `asRowSet()` needs to be closed.)
Or we can do this with a single `find`, a row sequence iterator, and position distances.
Or this way:
```
        final MutableLong position = new MutableLong(0);
        try (final RowSequence.Iterator rowSetIterator = rowSet.getRowSequenceIterator()) {
            rowSequence.forAllRowKeys(rowKey -> {
                position.add(rowSetIterator.advanceAndGetPositionDistance(rowKey));
                longChunk.add(position.longValue());
            });
        }
```

Reply:
I did some contextual testing (rather than explicit for any kind of input row set and filled row sequence). The testing favors the `invert` approach with a single-range optimization before dropping to rowsequence's fill chunk.